src/controllers/attendanceManagement/getAttandanceOfASessionByDate.js

```diff
@@ -144,12 +148,164 @@ const getAttendanceOfASessionByDate = async (req, res) => {
     let attendance = null;
     if (attendances.length >= 1) {
       attendance = attendances[0];
+    }else{
+      let startDate=req.sessionData.startDate;
+      let endDate=req.sessionData.endDate;
+      if(date>=startDate && date<endDate){
+        let enrolments = await Enrolment.find({sessionId:sessionId,enrolledStatus:'ENROLLED',startDate:{$lte: date }});
+        if(enrolments.length>0){
+          // create the record for the members
+          req.body.records=[];
+          for(let i=0;i<enrolments.length;i++){
+            let recordObj = {
+              'memberId':enrolments[i].memberId,
+              'attended':false,
+              'comment':''
+            }
+            req.body.records.push(recordObj);
+          }
+          await addAttendanceHandler(req, { session });
+          attendance = aggregateResponse(sessionId,date,classId,businessId,req);
```

Comment:
use the same aggregateResponse query in the above aggregate query. So that, If you need to change anything, You can change it in one place.